crates/syntax/src/ast/token_ext.rs

```diff
@@ -319,6 +319,30 @@ impl ast::IntNumber {
             Some(suffix)
         }
     }
+
+    pub fn float_value(&self) -> Option<FloatTypeWrapper> {
+        let (_, text, _) = self.split_into_parts();
+        let float_value = text.parse::<f64>().ok()?;
+        Some(FloatTypeWrapper::new(float_value))
+    }
+}
+
+// We convert float values into bits and that's how we don't need to deal with f32 and f64.
+// For PartialEq, bits comparison should work, as ordering is not important
+// https://github.com/rust-lang/rust-analyzer/issues/12380#issuecomment-1137284360
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct FloatTypeWrapper(u64);
+
+impl FloatTypeWrapper {
+    fn new(value: f64) -> Self {
+        Self(value.to_bits())
+    }
+}
+
+impl std::fmt::Display for FloatTypeWrapper {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", f64::from_bits(self.0))
+    }
```

Comment:
This shouldn't go into `syntax`, let's put this where it is used by hir-def

Reply:
Shifted it to same file where `Literal::Float` is defined